### PR TITLE
match .NET naming guidelines

### PR DESCRIPTION
**Pull request by henrygab**
The first commit corresponds to the original pull request.
The second commit corresponds to the fix to allow private fields to be named however you'd like.

### CODE REVIEW

**Comment by henrygab** [comment]
@RehanSaeed , this pull request should meet all your preferences.  I even added the custom rules per your request (those that go beyond the .NET guidelines).  So, your review and integration would be appreciated, so I can delete my fork(s).

Thank you!

File: src/EditorConfigSamples.csproj [hunk 1]
```diff
@@ -0,0 +1,68 @@
+﻿<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (RehanSaeed) [inline src/EditorConfigSamples.csproj]: Can we use a .NET Core project? All this XML with these old style `csproj` files projects hurts my eyes.

File: src/EditorConfigSamples.ruleset [hunk 2]
```diff
@@ -0,0 +1,79 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (RehanSaeed) [inline src/EditorConfigSamples.ruleset]: Hmmmm. Maybe we should provide a `.ruleset` file at the root that conforms? [This](https://github.com/Dotnet-Boxed/Templates/blob/master/MinimumRecommendedRulesWithStyleCop.ruleset) is the one I use everywhere. It turns off document everything rules but generally sticks to defaults. If we do this, we should give it a nice name and description.

File: src/stylecop.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (RehanSaeed) [inline src/stylecop.json]: If you use a `.ruleset` file, I don't think you need this.

File: src/EnumExamples.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// Copyright (c) 2019 Henry Gabryjelski
```
Comment (RehanSaeed) [inline src/EnumExamples.cs]: Do we have to enforce a copyright?

File: README.md [hunk 5]
```diff
@@ -17,7 +17,10 @@ A very generic [.editorconfig](https://github.com/RehanSaeed/EditorConfig/blob/m
 - Markdown - .md
 - Visual Studio - .sln, .csproj, .vbproj, .vcxproj, .vcxproj.filters, .proj, .projitems, .shproj
 
-Extensive code style settings for C# and VB.NET have been defined that require the latest C# features to be used. Also, all C# related code styles are consistent with [StyleCop's](https://github.com/DotNetAnalyzers/StyleCopAnalyzers) default styles.
+### For .NET code
+Extensive code style settings for C# and VB.NET have been defined that require the latest C# features to be used.
```
Comment (RehanSaeed) [inline README.md]: Exactly!

File: src/EditorConfigSamples.csproj (hunk 1 above)
Comment (henrygab) [reply src/EditorConfigSamples.csproj]: Feel free to convert or delete afterwards.

File: src/EditorConfigSamples.ruleset (hunk 2 above)
Comment (henrygab) [reply src/EditorConfigSamples.ruleset]: **_Feel free to edit or delete this project._**  My goal was simply to remove warnings not related to the issue at hand.

File: src/stylecop.json (hunk 3 above)
Comment (henrygab) [reply src/stylecop.json]: See prior comments.

File: src/EnumExamples.cs (hunk 4 above)
Comment (henrygab) [reply src/EnumExamples.cs]: Copyrights exist even if not marked.  License still applies.

**Comment by henrygab** [comment]
Upon request, I will delete the entire project.  Otherwise, let's keep the focus on the key topic, which is the .editorconfig file ….

**Comment by RehanSaeed** [comment]
Perhaps we can move it to another PR.

**Comment by henrygab** [comment]
That's fine.  For now, use the project as a basis for any remaining comments, if you fee there's anything left that's **_incorrect_** in the .editorconfig file, knowing that it will be removed before you approve the PR.

**Comment by RehanSaeed** [comment]
All good. My only query is about adding the MIT license into the `.editorconfig` file as it makes it a bit longer. The closest example to this project is the [Visual Studio .gitignore file](https://github.com/github/gitignore/blob/master/VisualStudio.gitignore) which doesn't include a license.

I found out that you work with Patents at Microsoft though, so I guess I'll defer to your superior knowledge about adding it in.

**Pull request merged**